src/app/services/authorization.service.ts

```diff
@@ -0,0 +1,54 @@
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { UserData } from "@shared/interfaces/userData.interface";
+import { BehaviorSubject, map, Observable } from "rxjs";
+import { QuizService } from "@services/quiz.service";
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthorizationService {
+
+  public isAuth$ = new BehaviorSubject<boolean>(false);
+  public username$ = new BehaviorSubject<string>('');
+  private _url = 'http://localhost:8080/api/auth/';
+  private _headers = new HttpHeaders({ 'Content-Type': 'application/json' });
+  private _jwtHelper = new JwtHelperService();
+
+  constructor(private http: HttpClient, private quizService: QuizService) {
+    const user = JSON.parse(sessionStorage.getItem('user')!);
```

Comment:
@KovalivNataliia IF user value is not changed, move it to general variables
Otherwise, move this block to a separate function for more readability.